src/Linters/LintError.hack

```diff
@@ -9,52 +9,28 @@
 
 namespace Facebook\HHAST;
 
-use type Facebook\HHAST\File;
-
-class LintError {
-  public function __construct(
-    private BaseLinter $linter,
-    private string $description,
-  ) {
-  }
-
-  final public function getFile(): File {
-    return $this->linter->getFile();
-  }
-
-  public function getPosition(): ?(int, int) {
-    return null;
-  }
-
-  public function getRange(): ?((int, int), ?(int, int)) {
-    $pos = $this->getPosition();
-    if ($pos === null) {
-      return null;
-    }
-    return tuple($pos, null);
-  }
-
-  public function getDescription(): string {
-    return $this->description;
-  }
-
-  /** Retrieve a snippet of the source code that is responsible for the error.
-   */
-  public function getBlameCode(): ?string {
-    return null;
-  }
-
-  /** Retrieve a human-friendly version of "what's responsible".
-   *
-   * For example, for "await in a loop", `getBlameCode()` may point at the
-   * await statement, but this function may return the entire loop, annotated
-   * with pointers.
-   */
-  public function getPrettyBlame(): ?string {
-    return $this->getBlameCode();
-  }
-
-  final public function getLinter(): BaseLinter {
-    return $this->linter;
-  }
+/**
+ * A problem detected by a lint rule
+ *
+ * LintError is more generic than SingleRuleLintError because LintError is associated with a
+ * LintRule, not a SingleRuleLinter.
+ */
+<<__Sealed(
+  SingleRuleLintError::class,
+  // HHClientProblem::class
+)>>
+interface LintError {
+  public function getPath(): string;
```

Comment:
This needs to return a `File`, as linters can be ran on stuff that doesn't exist on disk (or doesn't match), especially when the linters are being ran interactively while editing the file in an IDE